nixos/modules/services/misc/n8n.nix

```diff
@@ -0,0 +1,433 @@
+          };
+        };
+
+        jwtAuth = {
+          active = mkEnableOption "If jwt auth should be activated for editor and REST-API";
+
+          jwtHeader = mkOption {
+            type = types.str;
+            default = "";
+            description = "The request header containing a signed JWT";
+          };
+
+          jwtHeaderValuePrefix = mkOption {
+            type = types.str;
+            default = "";
+            description = "The request header value prefix to strip (optional)";
+          };
+
+          jwksUri = mkOption {
+            type = types.str;
+            default = "";
+            description = "The URI to fetch JWK Set for JWT authentication";
+          };
+
+          jwtIssuer = mkOption {
+            type = types.str;
+            default = "";
+            description = "JWT issuer to expect (optional)";
+          };
+
+          jwtNamespace = mkOption {
+            type = types.str;
+            default = "";
+            description = "JWT namespace to expect (optional)";
+          };
+
+          jwtAllowedTenantKey = mkOption {
+            type = types.str;
+            default = "";
+            description = "JWT tenant key name to inspect within JWT namespace (optional)";
+          };
+
+          jwtAllowedTenant = mkOption {
+            type = types.str;
+            default = "";
+            description = "JWT tenant to allow (optional)";
+          };
+        };
+      };
+
+      endpoints = {
+        rest = mkOption {
+          type = types.str;
+          default = "rest";
+          description = "Path for rest endpoint";
+        };
+
+        webhook = mkOption {
+          type = types.str;
+          default = "webhook";
+          description = "Path for webhook endpoint";
+        };
+
+        webhookTest = mkOption {
+          type = types.str;
+          default = "webhook-test";
+          description = "Path for test-webhook endpoint";
+        };
+      };
+
+      externalHookFiles = mkOption {
+        type = types.listOf types.str;
+        default = [];
+        description = "Files containing external hooks. Multiple files can be separated by colon (\":\")";
+      };
+
+      nodes = {
+        exclude = mkOption {
+          type = types.listOf types.str;
+          default = [];
+          description = "Nodes not to load";
+        };
+
+        errorTriggerType = mkOption {
+          type = types.str;
+          default = "n8n-nodes-base.errorTrigger";
+          description = "Node Type to use as Error Trigger";
+        };
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.n8n = {
+      description = "N8N service";
+      after = [ "target.target" ];
+      wantedBy = [ "multi-user.target" ];
+      serviceConfig = {
+        Type = "simple";
+        User = cfg.user;
+        Group = cfg.group;
+        StateDirectory = "n8n";
+        SyslogIdentifier = "n8n";
+        Environment = "N8N_CONFIG_FILES=${configFile}";
```

Comment:
Personally I prefer `systemd.services.n8n.environment.N8N_CONFIG_FILES` because it allows the user more control, but this is fine as well if you don't like my suggestion.

Reply:
No problem

Reply:
This isn't needed anymore.